repairs/utils.py

```diff
@@ -79,13 +79,18 @@ def set_iem_owner(warranty_claim, method):
 
 			if impression_id:
 				if frappe.db.exists("Serial No", serial_no):
-					iem_owner = frappe.db.get_value("IEM Owner", {"impression_id": impression_id}, "name")
-
 					frappe.db.set_value("Serial No", serial_no, "impression_id", impression_id)
-					frappe.db.set_value("Serial No", serial_no, "iem_owner", iem_owner)
+
+					iem_owner = frappe.get_all("IEM Owner", or_filters={"impression_id": impression_id, "old_impression_id": impression_id})
+					if iem_owner:
+						frappe.db.set_value("Serial No", serial_no, "iem_owner", iem_owner[0].name)
 
 		if impression_id:
-			warranty_claim.iem_owner = frappe.db.get_value("IEM Owner", {"impression_id": impression_id}, "name")
+			iem_owner = frappe.get_all("IEM Owner", or_filters={"impression_id": impression_id, "old_impression_id": impression_id})
```

Comment:
I see us using get_all all the time, in this case its probably not a problem but on places where it would fetch a lot of data and we are only interested on the first result we should add a limit=1 on the call to avoid frappe parsing as much as possible